cloud.ganttproject.colloboque/src/main/resources/init-colloboque-template-database.sql

```diff
@@ -208,39 +231,126 @@ CREATE TABLE TaskName(
 CREATE TABLE TaskDates(
     uid           TEXT PRIMARY KEY REFERENCES TaskName,
     start_date    DATE NOT NULL,
-    duration_days INT NOT NULL
+    duration_days INT NOT NULL DEFAULT 1,
+    earliest_start_date DATE
 );
 
 -- Other task properties can be changed independently, so they are stored in rows, one row corresponds to one
 -- instance of the task property value
 
-CREATE TYPE IntPropertyName AS ENUM ('completion', 'priority');
-
 -- Integer valued properties
 CREATE TABLE TaskIntProperties(
     uid        TEXT REFERENCES TaskName,
-    prop_name  IntPropertyName,
-    prop_value INT
+    prop_name  project_model_metadata.TaskIntPropertyName,
+    prop_value INT,
+    PRIMARY KEY(uid, prop_name)
+);
+
+-- Text valued properties
+CREATE TABLE TaskTextProperties(
+    uid TEXT REFERENCES TaskName,
+    prop_name project_model_metadata.TaskTextPropertyName,
+    prop_value TEXT,
+    PRIMARY KEY(uid, prop_name)
+);
+
+CREATE TABLE TaskCostProperties(
+    uid TEXT REFERENCES TaskName PRIMARY KEY,
+    is_cost_calculated BOOLEAN,
+    cost_manual_value NUMERIC
+);
+
+CREATE TABLE TaskClassProperties(
+    uid TEXT REFERENCES TaskName PRIMARY KEY,
+    is_milestone BOOLEAN NOT NULL DEFAULT false,
+    is_project_task BOOLEAN NOT NULL DEFAULT false
 );
 
 -- Updatable view which collects all task properties in a single row. Inserts and updates are processed
 -- with INSTEAD OF triggers.
 CREATE VIEW Task AS
-SELECT uid, num, name, start_date, duration_days AS duration,
-       MAX(prop_value) FILTER(WHERE prop_name = 'completion') AS completion
-from TaskName JOIN TaskDates USING(uid)
-         LEFT JOIN TaskIntProperties USING(uid)
-GROUP BY uid, TaskDates.uid;
+SELECT uid,
+       num,
+       name,
+       start_date,
+       duration_days AS duration,
+       earliest_start_date,
+       is_cost_calculated,
+       cost_manual_value,
+       is_milestone,
+       is_project_task,
+       MAX(TIP.prop_value) FILTER (WHERE TIP.prop_name = 'completion') AS completion,
```

Comment:
Max is nominal here to confirm that we select a single value, right?

Reply:
Yes, just to make SELECT+GROUP BY happy. In a group filtered by the property name we'll have no more than 1 value anyway.

Reply:
Alright, thanks!